kernel/syscall.c

```diff
@@ -3126,6 +3126,57 @@ long myst_syscall_mbind(
     return ret;
 }
 
+long myst_syscall_get_mempolicy(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    unsigned long rounded_maxnode = 0;
+    int ROUND_UP_VAL = sizeof(unsigned long) * 8;
+    myst_round_up(maxnode, ROUND_UP_VAL, &rounded_maxnode);
+
+    unsigned long* rounded_nodemask;
+    // allocate higher buffer
+    if (rounded_maxnode > maxnode)
+    {
+        rounded_nodemask = malloc(rounded_maxnode / CHAR_BIT);
+    }
+    else
+    {
+        rounded_nodemask = nodemask;
+    }
+
```

Comment:
zero out `nodemask` before calling lower level function, in case the app code didn't do that.